spring-cloud-azure-config/src/main/java/com/microsoft/azure/spring/cloud/config/resource/ConnectionString.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+package com.microsoft.azure.spring.cloud.config.resource;
+
+import org.springframework.util.Assert;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class ConnectionString {
+    private static final String CONN_STRING_REGEXP = "Endpoint=https://([^;]+);Id=([^;]+);Secret=([^;]+)";
+    public static final String ENDPOINT_ERR_MSG = String.format("Connection string does not follow format %s.",
+            CONN_STRING_REGEXP);
+    private static final Pattern CONN_STRING_PATTERN = Pattern.compile(CONN_STRING_REGEXP);
+    public static final String NON_EMPTY_MSG = "%s property should not be null or empty in the connection string of " +
+            "Azure Config Service.";
+
+    private String endpoint;
+    private String id;
+    private String secret;
+
+    public ConnectionString(String endpoint, String id, String secret) {
```

Comment:
If no direct call of this constructor, better be `private`.

Reply:
Make sense, for now there is no external access yet, use private first. Updated.